dev/io.openliberty.accesslists.internal/src/io/openliberty/accesslists/GenericAccessLists.java

```diff
@@ -0,0 +1,128 @@
+/*******************************************************************************
+ * Copyright (c) 2021 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.accesslists;
+
+import java.net.Inet6Address;
+import java.net.InetAddress;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.ws.tcpchannel.internal.FilterList;
+import com.ibm.ws.tcpchannel.internal.FilterListStr;
+import com.ibm.ws.udpchannel.internal.AccessLists;
+
+import io.openliberty.accesslists.AccessListKeysFacade;
+import io.openliberty.accesslists.AccessListsConstants;
+
+/**
+ * Wrapper class handling various include or exclude lists for accessing an
+ * inbound port.
+ */
+public class GenericAccessLists {
+    
+    public static final TraceComponent tc = Tr.register(GenericAccessLists.class, AccessListsConstants.TCP_TRACE_GROUP,
+            AccessListsConstants.TCP_MESSAGES);
+    
+    public FilterList excludeAccess = null;
+    public FilterList includeAccess = null;
+
+    public GenericAccessLists(FilterList _excludeAccess, FilterList _includeAccess) {
+        this.excludeAccess = _excludeAccess;
+        this.includeAccess = _includeAccess;
+    }
+
+    /**
+     * @param addr
+     * @return Will return true if there is an access include list and the address is not on it or 
+     * (lazy or) the address exclude list is not empty and the address is on it
+     */
+    public boolean accessDenied(InetAddress addr) {
+        return nonEmptyIncludesAndNotFound(addr) || inAddressExcludes(addr);
+    }
+
+    /**
+     * Return true if the addressIncludeList is non-empty but the address is not in it
+     * 
+     * @param addr
+     * @return true if there is an include list present but the address is not in it
+     */
+    protected boolean nonEmptyIncludesAndNotFound(InetAddress addr) {
+
+        if (!includeAccess.getActive()) {
```

Comment:
can we get rid of this check since it's duplicated in `inAddressIncludes()`?

Reply:
As discussed above, in order to have the largest chunk of common code for inAddressInclude - I need to have this method mean 'excluded due to includes' so that I can override it for TCP over UDP to include Hosts - it can't just be found in one() || other() with embedded isActive checks in that case.

Reply:
Thanks again for the clarification in you comment in `AccessLists`, this is good with me.